skimage/restoration/tests/test_rolling_ball.py

```diff
@@ -18,6 +19,15 @@ def test_ellipsoid_const():
     assert np.allclose(img - background, np.zeros_like(img))
 
 
+@testing.parametrize('dtype', [np.float32, np.float64])
+def test_rolling_ball_dtype(dtype):
+    img = 155 * np.ones((100, 100), dtype=dtype)
+    kernel = ellipsoid_kernel((25, 53), 50).astype(dtype, copy=False)
+    background = rolling_ball(img, kernel=kernel)
+    assert background.dtype == img.dtype
+    assert np.allclose(img - background, np.zeros_like(img))
```

Comment:
while it's not wrong, it's a bit unexpected to test for numerical correctness in a function that claims to test datatypes. If it fails because of this line, one might be lead down the wrong path and search for a ´dtype´ issue.